packages/flutter_tools/lib/src/asset.dart

```diff
@@ -1065,6 +1181,21 @@ class _Asset {
     return index == -1 ? null : Uri(path: entryUri.path.substring(0, index));
   }
 
+  bool matchesFlavor(String? flavor) {
+    if (flavors.isEmpty) {
+      return true;
+    }
+
+    return flavors.any((String e) => e == flavor);
```

Comment:
today it matches nothing. which is better than matching everything, assuming we reached this line because `flavors.isNotEmpty`. is it valid to have flavors configured on a project and not pass one? and if not, how do we handle that case?

Reply:
> is it valid to have flavors configured on a project and not pass one?

Great question. If I had to guess, the behavior for this hasn't been defined. Suppose you have a Flutter project with a `free` flavor configured. If you try to run/build without `--flavor` on Android, you'll see this error similar to this:
```
Gradle build failed to produce an .apk file. It's likely that this file was generated under /Users/andrewkolos/Desktop/deleteme/throwaway/build, but the tool couldn't find it.
```

However, you can build/run against iOS without `--flavor` without issue. 🤷 

To return on the root question:

> Also what does a null flavor imply. Should that match everything?

Yes, it's intended that a flavorless asset matches any flavor, including the lack of a flavor (`null`). I've made this more explicit by adding another early return. https://github.com/flutter/flutter/blob/f7e61ebf295dc530a8199d220bb99f03c1a9ad19/packages/flutter_tools/lib/src/asset.dart#L1184-L1194